### ISSUE TITLE
[JIT] elementwise min/max attempting to create a Scalar from a 3 dim Tensor

### ISSUE THREAD

**Issue by jamesr66a** [pytorch/pytorch#8513]
Operator level tests fail with some error about creating a Scalar from a 3-dim tensor with elementwise `max` and `min`. Example failure:

```
======================================================================
ERROR: test_max_elementwise (__main__.TestAutogradGenerated)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test/test_autograd.py", line 3665, in do_test
    check(name)
  File "test/test_autograd.py", line 3592, in check
    fn, (self_variable,) + args_variable)
  File "test/test_autograd.py", line 3498, in check_against_reference
    outputs_test = func(*nograd_inputs)
  File "test/test_autograd.py", line 3474, in script_fn
    return output_process_fn(CU.the_method(*tensors))
RuntimeError: 
Attempting to create a Scalar from a 3 dim tensor (Scalar at /data/users/zdevito/pytorch/torch/lib/tmp_install/include/ATen/Scalar.h:27)
frame #0: <unknown function> + 0x570380 (0x7fa6c2021380 in /data/users/zdevito/pytorch/torch/_C.so)
frame #1: torch::jit::InterpreterStateImpl::runOneStage(std::vector<at::Tensor, std::allocator<at::Tensor> >&) + 0x124 (0x7fa6c1e64914 in /data/users/zdevito/pytorch/torch/_C.so)
frame #2: torch::jit::GraphExecutorImpl::run(torch::jit::variable_tensor_list) + 0xbb6 (0x7fa6c1e860f6 in /data/users/zdevito/pytorch/torch/_C.so)
frame #3: torch::jit::GraphExecutor::run(torch::jit::variable_tensor_list&&) + 0x47 (0x7fa6c1e803c7 in /data/users/zdevito/pytorch/torch/_C.so)
frame #4: <unknown function> + 0x62ce29 (0x7fa6c20dde29 in /data/users/zdevito/pytorch/torch/_C.so)
frame #5: <unknown function> + 0x31af88 (0x7fa6c1dcbf88 in /data/users/zdevito/pytorch/torch/_C.so)
frame #6: PyObject_Call + 0x53 (0x7fa6d1efce93 in /data/users/zdevito/anaconda2/bin/../lib/libpython2.7.so.1.0)
frame #7: <unknown function> + 0x5c64f (0x7fa6d1f0f64f in /data/users/zdevito/anaconda2/bin/../lib/libpython2.7.so.1.0)
frame #8: PyObject_Call + 0x53 (0x7fa6d1efce93 in /data/users/zdevito/anaconda2/bin/../lib/libpython2.7.so.1.0)
frame #9: <unknown function> + 0xba2ac (0x7fa6d1f6d2ac in /data/users/zdevito/anaconda2/bin/../lib/libpython2.7.so.1.0)
frame #10: PyObject_Call + 0x53 (0x7fa6d1efce93 in /data/users/zdevito/anaconda2/bin/../lib/libpython2.7.so.1.0)
frame #11: PyEval_EvalFrameEx + 0x61d6 (0x7fa6d1fae886 in /data/users/zdevito/anaconda2/bin/../lib/libpython2.7.so.1.0)
frame #12: PyEval_EvalCodeEx + 0x89e (0x7fa6d1fb1c3e in /data/users/zdevito/anaconda2/bin/../lib/libpython2.7.so.1.0)
frame #13: <unknown function> + 0x79a61 (0x7fa6d1f2ca61 in /data/users/zdevito/anaconda2/bin/../lib/libpython2.7.so.1.0)
frame #14: PyObject_Call + 0x53 (0x7fa6d1efce93 in /data/users/zdevito/anaconda2/bin/../lib/libpython2.7.so.1.0)
frame #15: PyEval_EvalFrameEx + 0x61d6 (0x7fa6d1fae886 in /data/users/zdevito/anaconda2/bin/../lib/libpython2.7.so.1.0)
frame #16: PyEval_EvalCodeEx + 0x89e (0x7fa6d1fb1c3e in /data/users/zdevito/anaconda2/bin/../lib/libpython2.7.so.1.0)
frame #17: PyEval_EvalFrameEx + 0x8b47 (0x7fa6d1fb11f7 in /data/users/zdevito/anaconda2/bin/../lib/libpython2.7.so.1.0)
frame #18: PyEval_EvalCodeEx + 0x89e (0x7fa6d1fb1c3e in /data/users/zdevito/anaconda2/bin/../lib/libpython2.7.so.1.0)
frame #19: PyEval_EvalFrameEx + 0x8b47 (0x7fa6d1fb11f7 in /data/users/zdevito/anaconda2/bin/../lib/libpython2.7.so.1.0)
frame #20: PyEval_EvalCodeEx + 0x89e (0x7fa6d1fb1c3e in /data/users/zdevito/anaconda2/bin/../lib/libpython2.7.so.1.0)
frame #21: PyEval_EvalFrameEx + 0x8b47 (0x7fa6d1fb11f7 in /data/users/zdevito/anaconda2/bin/../lib/libpython2.7.so.1.0)
frame #22: PyEval_EvalCodeEx + 0x89e (0x7fa6d1fb1c3e in /data/users/zdevito/anaconda2/bin/../lib/libpython2.7.so.1.0)
frame #23: <unknown function> + 0x79b68 (0x7fa6d1f2cb68 in /data/users/zdevito/anaconda2/bin/../lib/libpython2.7.so.1.0)
frame #24: PyObject_Call + 0x53 (0x7fa6d1efce93 in /data/users/zdevito/anaconda2/bin/../lib/libpython2.7.so.1.0)
frame #25: PyEval_EvalFrameEx + 0x61d6 (0x7fa6d1fae886 in /data/users/zdevito/anaconda2/bin/../lib/libpython2.7.so.1.0)
frame #26: PyEval_EvalCodeEx + 0x89e (0x7fa6d1fb1c3e in /data/users/zdevito/anaconda2/bin/../lib/libpython2.7.so.1.0)
frame #27: <unknown function> + 0x79a61 (0x7fa6d1f2ca61 in /data/users/zdevito/anaconda2/bin/../lib/libpython2.7.so.1.0)
frame #28: PyObject_Call + 0x53 (0x7fa6d1efce93 in /data/users/zdevito/anaconda2/bin/../lib/libpython2.7.so.1.0)
frame #29: <unknown function> + 0x5c64f (0x7fa6d1f0f64f in /data/users/zdevito/anaconda2/bin/../lib/libpython2.7.so.1.0)
frame #30: PyObject_Call + 0x53 (0x7fa6d1efce93 in /data/users/zdevito/anaconda2/bin/../lib/libpython2.7.so.1.0)
frame #31: <unknown function> + 0xba2ac (0x7fa6d1f6d2ac in /data/users/zdevito/anaconda2/bin/../lib/libpython2.7.so.1.0)
frame #32: PyObject_Call + 0x53 (0x7fa6d1efce93 in /data/users/zdevito/anaconda2/bin/../lib/libpython2.7.so.1.0)
frame #33: PyEval_EvalFrameEx + 0x715d (0x7fa6d1faf80d in /data/users/zdevito/anaconda2/bin/../lib/libpython2.7.so.1.0)
frame #34: PyEval_EvalCodeEx + 0x89e (0x7fa6d1fb1c3e in /data/users/zdevito/anaconda2/bin/../lib/libpython2.7.so.1.0)
frame #35: <unknown function> + 0x79b68 (0x7fa6d1f2cb68 in /data/users/zdevito/anaconda2/bin/../lib/libpython2.7.so.1.0)
frame #36: PyObject_Call + 0x53 (0x7fa6d1efce93 in /data/users/zdevito/anaconda2/bin/../lib/libpython2.7.so.1.0)
frame #37: PyEval_EvalFrameEx + 0x61d6 (0x7fa6d1fae886 in /data/users/zdevito/anaconda2/bin/../lib/libpython2.7.so.1.0)
frame #38: PyEval_EvalCodeEx + 0x89e (0x7fa6d1fb1c3e in /data/users/zdevito/anaconda2/bin/../lib/libpython2.7.so.1.0)
frame #39: <unknown function> + 0x79a61 (0x7fa6d1f2ca61 in /data/users/zdevito/anaconda2/bin/../lib/libpython2.7.so.1.0)
frame #40: PyObject_Call + 0x53 (0x7fa6d1efce93 in /data/users/zdevito/anaconda2/bin/../lib/libpython2.7.so.1.0)
frame #41: <unknown function> + 0x5c64f (0x7fa6d1f0f64f in /data/users/zdevito/anaconda2/bin/../lib/libpython2.7.so.1.0)
frame #42: PyObject_Call + 0x53 (0x7fa6d1efce93 in /data/users/zdevito/anaconda2/bin/../lib/libpython2.7.so.1.0)
frame #43: <unknown function> + 0xba2ac (0x7fa6d1f6d2ac in /data/users/zdevito/anaconda2/bin/../lib/libpython2.7.so.1.0)
frame #44: PyObject_Call + 0x53 (0x7fa6d1efce93 in /data/users/zdevito/anaconda2/bin/../lib/libpython2.7.so.1.0)
frame #45: PyEval_EvalFrameEx + 0x715d (0x7fa6d1faf80d in /data/users/zdevito/anaconda2/bin/../lib/libpython2.7.so.1.0)
frame #46: PyEval_EvalCodeEx + 0x89e (0x7fa6d1fb1c3e in /data/users/zdevito/anaconda2/bin/../lib/libpython2.7.so.1.0)
frame #47: <unknown function> + 0x79b68 (0x7fa6d1f2cb68 in /data/users/zdevito/anaconda2/bin/../lib/libpython2.7.so.1.0)
frame #48: PyObject_Call + 0x53 (0x7fa6d1efce93 in /data/users/zdevito/anaconda2/bin/../lib/libpython2.7.so.1.0)
frame #49: PyEval_EvalFrameEx + 0x61d6 (0x7fa6d1fae886 in /data/users/zdevito/anaconda2/bin/../lib/libpython2.7.so.1.0)
frame #50: PyEval_EvalCodeEx + 0x89e (0x7fa6d1fb1c3e in /data/users/zdevito/anaconda2/bin/../lib/libpython2.7.so.1.0)
frame #51: <unknown function> + 0x79a61 (0x7fa6d1f2ca61 in /data/users/zdevito/anaconda2/bin/../lib/libpython2.7.so.1.0)
frame #52: PyObject_Call + 0x53 (0x7fa6d1efce93 in /data/users/zdevito/anaconda2/bin/../lib/libpython2.7.so.1.0)
frame #53: <unknown function> + 0x5c64f (0x7fa6d1f0f64f in /data/users/zdevito/anaconda2/bin/../lib/libpython2.7.so.1.0)
frame #54: PyObject_Call + 0x53 (0x7fa6d1efce93 in /data/users/zdevito/anaconda2/bin/../lib/libpython2.7.so.1.0)
frame #55: <unknown function> + 0xba2ac (0x7fa6d1f6d2ac in /data/users/zdevito/anaconda2/bin/../lib/libpython2.7.so.1.0)
frame #56: PyObject_Call + 0x53 (0x7fa6d1efce93 in /data/users/zdevito/anaconda2/bin/../lib/libpython2.7.so.1.0)
frame #57: PyEval_EvalFrameEx + 0x715d (0x7fa6d1faf80d in /data/users/zdevito/anaconda2/bin/../lib/libpython2.7.so.1.0)
frame #58: PyEval_EvalFrameEx + 0x8c95 (0x7fa6d1fb1345 in /data/users/zdevito/anaconda2/bin/../lib/libpython2.7.so.1.0)
frame #59: PyEval_EvalFrameEx + 0x8c95 (0x7fa6d1fb1345 in /data/users/zdevito/anaconda2/bin/../lib/libpython2.7.so.1.0)
frame #60: PyEval_EvalCodeEx + 0x89e (0x7fa6d1fb1c3e in /data/users/zdevito/anaconda2/bin/../lib/libpython2.7.so.1.0)
frame #61: <unknown function> + 0x79b68 (0x7fa6d1f2cb68 in /data/users/zdevito/anaconda2/bin/../lib/libpython2.7.so.1.0)
frame #62: PyObject_Call + 0x53 (0x7fa6d1efce93 in /data/users/zdevito/anaconda2/bin/../lib/libpython2.7.so.1.0)
frame #63: <unknown function> + 0x5c64f (0x7fa6d1f0f64f in /data/users/zdevito/anaconda2/bin/../lib/libpython2.7.so.1.0)
:
operation failed in interpreter:

def the_method(i0, i1):
    return i0.max(i1)
           ~~~~~~ <--- HERE
```

Affected tests:
- `test_max_elementwise`
- `test_max_elementwise_broadcast_all`
- `test_max_elementwise_broadcast_lhs`
- `test_max_elementwise_broadcast_rhs`
- `test_max_scalar_elementwise_broadcast_lhs`
- `test_min_elementwise`
- `test_min_elementwise_broadcast_all`
- `test_min_elementwise_broadcast_lhs`
- `test_min_elementwise_broadcast_rhs`
- `test_min_scalar_elementwise_broadcast_lhs`

**Comment by jamesr66a** [comment]
@zou3519 assigning this to you because I think your changes will fix this. This is an error in schema matching for the `min` and `max` operators

**Comment by zou3519** [comment]
Yes, all of these are now fixed with #8687
